### PR TITLE
[ Tests ] Use Awaitility to instead of Thread sleep

### PR DESCRIPTION
**Pull request by mattisonchao**
### Motivation

Use Awaitility to instead of Thread sleep

### Modifications

- Use Awaitility to instead of Thread sleep

### CODE REVIEW

**Review by aahmed-se — APPROVED** [review]

**Review by sijie — APPROVED** [review]

File: pulsar-client-tools-test/src/test/java/org/apache/pulsar/client/cli/PulsarClientToolTest.java [hunk 1]
```diff
@@ -207,7 +186,7 @@ public void testDurableSubscribe() throws Exception {
         Assert.assertFalse(future.isCompletedExceptionally());
         future.get();
         //wait for close
-        Thread.sleep(2000);
+        Awaitility.waitAtMost(Duration.ofDays(2000));
```
Comment (lhotari) [inline pulsar-client-tools-test/src/test/java/org/apache/pulsar/client/cli/PulsarClientToolTest.java:189]: Was this change intentional?

Comment (mattisonchao) [reply pulsar-client-tools-test/src/test/java/org/apache/pulsar/client/cli/PulsarClientToolTest.java:189]: I'm so sorry, This is my mistake, I will fix it ASAP.

Comment (mattisonchao) [reply pulsar-client-tools-test/src/test/java/org/apache/pulsar/client/cli/PulsarClientToolTest.java:189]: #11176 FIxed this mistake
